integration/src/womir_translation.rs

```diff
@@ -819,39 +824,170 @@ impl<'a, F: PrimeField32> Settings<'a> for OpenVMSettings<F> {
                 }
             }
             Err(op) => op,
         };
 
-        // Handle the remaining operations
+        // Handle the complex instructions that supports constant inlining
+        match &op {
+            Op::I32GeS
+            | Op::I32GeU
+            | Op::I64GeS
+            | Op::I64GeU
+            | Op::I32LeS
+            | Op::I32LeU
+            | Op::I64LeS
+            | Op::I64LeU => {
+                let inverse_result = c.register_gen.allocate_type(ValType::I32).start as usize;
+                let output = output.unwrap().start as usize;
+
+                let comp_instruction = match [&inputs[0], &inputs[1]] {
+                    [WasmOpInput::Register(input1), WasmOpInput::Register(input2)] => {
+                        let inverse_op = match op {
+                            Op::I32GeS => ib::lt_s,
+                            Op::I32GeU => ib::lt_u,
+                            Op::I64GeS => ib::lt_s_64,
+                            Op::I64GeU => ib::lt_u_64,
+                            Op::I32LeS => ib::gt_s,
+                            Op::I32LeU => ib::gt_u,
+                            Op::I64LeS => ib::gt_s_64,
+                            Op::I64LeU => ib::gt_u_64,
+                            _ => unreachable!(),
+                        };
+
+                        let input1 = input1.start as usize;
+                        let input2 = input2.start as usize;
+
+                        // Perform the inverse operation and invert the result
+                        inverse_op(inverse_result, input1, input2)
+                    }
+                    [WasmOpInput::Register(reg), WasmOpInput::Constant(c)]
+                    | [WasmOpInput::Constant(c), WasmOpInput::Register(reg)] => {
+                        let opcode = match op {
+                            Op::I32GeS | Op::I32LeS => LessThanOpcode::SLT.global_opcode(),
+                            Op::I32GeU | Op::I32LeU => LessThanOpcode::SLTU.global_opcode(),
+                            Op::I64GeS | Op::I64LeS => LessThan64Opcode::SLT.global_opcode(),
+                            Op::I64GeU | Op::I64LeU => LessThan64Opcode::SLTU.global_opcode(),
+                            _ => unreachable!(),
+                        };
+
+                        // Check whether this is the case where "c >= r" or "r <= c" can be turned into "r < c + 1".
+                        if ((matches!(op, Op::I32GeS | Op::I32GeU | Op::I64GeS | Op::I64GeU)
+                            && matches!(&inputs[0], WasmOpInput::Constant(_)))
+                            || (matches!(op, Op::I32LeS | Op::I32LeU | Op::I64LeS | Op::I64LeU)
+                                && matches!(&inputs[0], WasmOpInput::Register(_))))
+                            && let Some(inc_c) = can_turn_to_lt(&op, c)
+                        {
+                            return Directive::Instruction(ib::instr_i(
```

Comment:
minor: maybe add a small comment saying "no need to invert the result in this case"?

Reply:
But it is implicitly inverted, because of `inputs[0]` is matched to `Constant` in GE case, and to `Register` in LE case.

Maybe it would be clearer if instead of  `matches!(&inputs[0], WasmOpInput::Register(_))` I use  `matches!(&inputs[1], WasmOpInput::Constant(_))` ?

Reply:
i meant more in the sense that this is the only early return (avoiding the need for an extra instruction inverting the result)

Reply:
fixed